### DISCUSSION TITLE
Skip getServerSideProps on client routing

### DISCUSSION THREAD

**Discussion by isBatak** [vercel/next.js#11386] [Help]
Is it possible to skip `getServerSideProps` when navigating from one page to another on the client?
I want to prerender a page (SSR) when the user lands directly to the URL but when he navigates via client navigation I want to use `useSWR` and show the page with some skeleton loaders. 
Is that currently possible?

**Comment by aralroca** [comment]
I have exactly the same behavior under `getInitialProps` and I'm also looking for a solution before migrate from `getInitialProps` to `getServerSideProps`. 

Anybody knows if there is any way or any plan to support that?  

A suggestion: add some option to the return:


```js
async function getServerSideProps() {
  return {
    props: fetchProps(),
    navigation: false,
  }
}
```

**Reply by aralroca** [reply]
And perhaps, to don't rimplement then the fetch in a hook, It can return the status: pending, success, error, and the final props when is succeded.

**Reply by isBatak** [reply]
Yes, I was thinking about the same thing, some boolean value returned from `getServerSideProps` :D

**Reply by markbrocato** [reply]
We'd like to see this as well.  If we went ahead and sent a PR to implement this spec ^^, would it be accepted?  Is this already in progress.  Happy to help if no one else is working on it.

**Reply by timneutkens** [reply]
Open feature request here: https://github.com/zeit/next.js/issues/11897
You can't open a PR with that approach because it wouldn't work (as outlined in my replies on that feature request).

**Reply by kamilml** [reply]
I have exactly the same problem. I want to skip `getServerSideProps` in the client. Any solution or new idea? I'm getting crazy with this :(

I already spent 1 whole day trying to read my redux store (because I can save a value to prevent a useless query inside `getServerSideProps`) but I can't.

**Reply by thiagoloddi** [reply]
Is it just me or running `getServerSideProps` every time you change pages kind of defeats the purpose of a SPA? I want to make use of `apollo-client` in-memory cache but it's not possible, because whenever I change pages all the data is fetched again from the network in the server-side.

Is there a reason why it's implemented this way? I thought the option `shallow` should apply for changing routes within the app, not only when in the same page.

**Comment by leerob** [comment]
https://github.com/vercel/next.js/discussions/29117#discussioncomment-6870762
